src/js/components/Notification/Notification.js

```diff
@@ -17,6 +17,32 @@ import { Text } from '../Text';
 import { NotificationType } from './propTypes';
 import { useThemeValue } from '../../utils/useThemeValue';
 
+const MessageComponent = ({
+  message,
+  actions,
+  theme,
+  direction,
+  messageColor,
+}) => {
+  const Message = direction === 'row' ? Text : Paragraph;
+
+  const messageFill =
+    Message === Paragraph
+      ? theme.notification.message?.fill || false
+      : undefined;
+
+  return (
+    <Message
+      {...theme.notification.message}
+      color={messageColor}
+      fill={messageFill}
+    >
+      <Text margin={{ right: 'xsmall' }}>{message}</Text>
+      {actions}
+    </Message>
+  );
+};
+
```

Comment:
I think you could simplify to this (plus the suggestion where it's used).
The fill prop should end up getting passed in from the fact that theme.notification.message is already spread into the props. This versions also avoids passing fill to Text. And it avoids making the paragraph or text decision twice (once for the component and once for the fill.) Also by just letting the children come in via `rest` it avoids needing the `message`, and `actions` parameters (or similarly `color`), keeping it more agnostic. I'd considered renaming direction to a different property name since this really doesn't care about direction but rather something telling it to use Paragraph. But this is probably fine.
```suggestion
const Message = ({
  fill,
  direction,
  ...rest,
}) => {
  return direction === 'row'
    ? <Text {...rest} />
    : <Paragraph {...rest} fill={fill || false} />;
};
```

Reply:
Im going to leave the name `direction` as is just because we have in the theme `notification.direction` 
just so we dont get confused I did think we could change it to `messageLayout` but going to leave off for now since that also might confuse.